reviews/models/mixins.py

```diff
@@ -171,9 +175,33 @@ def save_changes(self, ev):
         now = self.action.date_created if self.action is not None else timezone.now()
         should_publish = self.reviewable.in_public_reviews_state
         if should_publish and not self.reviewable.is_published:
-            self.reviewable.is_published = True
+            if not (self.reviewable.node.preprint_file and self.reviewable.node.preprint_file.node == self.reviewable.node):
+                raise ValueError('Preprint node is not a valid preprint; cannot publish.')
+            if not self.reviewable.provider:
+                raise ValueError('Preprint provider not specified; cannot publish.')
+            if not self.reviewable.subjects.exists():
+                raise ValueError('Preprint must have at least one subject to be published.')
             self.reviewable.date_published = now
-            # TODO EZID (MOD-70)
+            self.reviewable.node._has_abandoned_preprint = False
+            self.reviewable.is_published = True
+            user = ev.kwargs.get('user')
+            auth = Auth(user)
+            self.reviewable.node.add_log(
+                action=NodeLog.PREPRINT_INITIATED,
+                params={
+                    'preprint': self.reviewable._id
+                },
+                auth=auth,
+                save=False,
+            )
+            if not self.reviewable.node.is_public:
+                self.node.set_privacy(
```

Comment:
This should not make the node public. If the node is private, trying to act on the preprint should fail. That would actually be a good check to add at the top of `save_changes`, and raise an exception.